### PR TITLE
Have `fct_lump()` throw error if `n` or `prop` not numeric

### LINKED ISSUE #169: Inconsistent behavior for fct_lump() when n or prop is a character
While `fct_lump()` keeps all the levels if you set `n` equal to a character, it sets them all to other when you set `prop` equal to a character.

Reprex: 

``` r
library(forcats)
x <- factor(letters[rpois(10, 5)])
# this keeps all the levels
fct_lump(x, n = "a")
#>  [1] c c b f e c b c n h
#> Levels: b c e f h n

# this makes everything "Other"
fct_lump(x, prop = "a")
#>  [1] Other Other Other Other Other Other Other Other Other Other
#> Levels: Other
```

<sup>Created on 2019-01-19 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

My personal opinion is that these should both result in an error instead, but I think they should at least be consistent.

### PR DESCRIPTION
**Pull request by robinsones**
Fixes #169. Also adds tests and NEWS item.

### CODE REVIEW

**Comment by robinsones** [comment]
FYI I noticed when I did this that `devtools::document()` hadn't been done after the change to `fct_cross()` and when it is there's a warning in `devtools::check()`: 

```
> checking Rd \usage sections ... WARNING
  Documented arguments not in \usage in documentation object 'fct_cross':
    ‘.f’
  
  Functions with \usage entries need to have the appropriate \alias
  entries, and all their arguments documented.
  The \usage entries must correspond to syntactically valid R code.
  See chapter ‘Writing R documentation files’ in the ‘Writing R
  Extensions’ manual.

0 errors ✓ | 1 warning x | 0 notes ✓
```

File: R/lump.R [hunk 1]
```diff
@@ -72,6 +72,9 @@ fct_lump <- function(f, n, prop, w = NULL, other_level = "Other",
   if (!xor(missing(n), missing(prop))) {
     new_levels <- ifelse(!in_smallest(count), levels, other_level)
   } else if (!missing(n)) {
+    if (!is.numeric(n)) {
+      stop("`n` must be numeric")
```
Comment (hadley) [inline R/lump.R]: Can you please use `rlang::abort()` here? (and below)

File: tests/testthat/test-fct_lump.R [hunk 2]
```diff
@@ -128,6 +128,14 @@ test_that("fct_lump_min works when weighted", {
   expect_equal(levels(fct_lump_min(f, min = 1.5, w = w)), c("b", "c", "d", "Other"))
 })
 
+test_that("throws error if n or prop is not numeric", {
+  f <- c("a", "a", "a", "a", "b", "b", "b", "c", "c", "d")
+
+  expect_error(fct_lump(f, n = "2"))
```
Comment (hadley) [inline tests/testthat/test-fct_lump.R]: Can you please add a "n" and "prop" to the test, just to match sure that you're catching the error that you think you are?

File: R/lump.R (hunk 1 above)
Comment (robinsones) [reply R/lump.R]: done

File: tests/testthat/test-fct_lump.R (hunk 2 above)
Comment (robinsones) [reply tests/testthat/test-fct_lump.R]: done

**Comment by hadley** [comment]
Thanks @robinsones!

**Pull request merged**